cartoframes/context.py

```diff
@@ -1614,17 +1614,21 @@ def data(self, table_name, metadata, persist_as=None, how='the_geom'):
             **DEFAULT_SQL_ARGS
         )['fields'].keys()
 
-        if set(tablecols) & set(_meta['suggested_name']):
-            commoncols = set(tablecols) & set(_meta['suggested_name'])
-            raise NameError('Column name collision for column(s): {cols}. '
-                            'Rename table column(s) to resolve.'.format(
-                                cols=', '.join(commoncols)))
+        names = {}
+        for row in _meta.iterrows():
```

Comment:
I wonder if iterating over `_meta['suggested_name']` would be faster than iterating over the entire dataframe. My guess is that it would be faster because it copies less data on every loop, but I'm not sure.

```python
for col in _meta['suggested_name']:
   ...
```

Reply:
Without knowing anything about the internals, I guess that it might be slower, but as it's iterated with `_meta.iterrows())` later I guess that it might cache internally, so the total time is not truly affected.

Anyway, your suggestion is not only a probable performance improvement but also more readable, as it focuses on the only column that is actually needed, so changing!